modules/tobira/src/main/java/org/opencastproject/tobira/impl/Item.java

```diff
@@ -264,41 +275,35 @@ private static List<Jsons.Val> findCaptions(MediaPackage mp) {
 
           return isVTT && isCorrectType;
         })
-        .forEach(track -> {
-          final var uri = track.getURI().toString();
-          captions.putIfAbsent(uri, new HashSet<String>());
-
-          // Try to get language from flavor. Otherwise there is an empty list
-          // in the map.
-          final var subflavor = track.getFlavor().getSubtype();
-          if (subflavor.startsWith("vtt+")) {
-            final var suffix = subflavor.substring("vtt+".length());
-            if (suffix.length() > 0) {
-              captions.get(uri).add(suffix);
+        .map(track -> {
+          final var tags = track.getTags();
+          final Function<String, Optional<String>> findTag = (String prefix) -> Arrays.stream(tags)
+                .filter(tag -> tag.startsWith(prefix))
+                .map(tag -> tag.substring(prefix.length()))
+                .findFirst();
```

Comment:
This doesn't actually work bc `generator` and `generator-type` have the same beginning and this also leaves in the ":".

```suggestion
          final Function<String, Optional<String>> findTag = (String prefix) -> Arrays.stream(tags)
                .map(tag -> tag.split(":"))
                .filter(tagArray -> (tagArray.length == 2 && tagArray[0].equals(prefix)))
                .map(tagArray -> tagArray[1])
                .findFirst();
```

Reply:
Woops :D Should be fixed